astropy/coordinates/earth.py

```diff
@@ -236,9 +236,9 @@ def from_geocentric(cls, x, y, z, unit=None):
         if unit is None:
             try:
                 unit = x.unit
-            except AttributeError:
+            except AttributeError as err:
                 raise TypeError("Geocentric coordinates should be Quantities "
-                                "unless an explicit unit is given.")
+                                "unless an explicit unit is given.") from err
```

Comment:
As I understand the exception raised is of the type`TypeError` - are you suggesting that the exception raised should be an `AttributeError` instead?

Reply:
It should definitely be TypeError. I was suggesting erasing the exception history using ``from None`` instead of ``from err``. Currently it will keep the error trace and say something like

TypeError
raised by Attributerror

But we're not interested in the AttributeError, that seems just an implementation detail to detect a quantity.